<launch>
  <!-- Arguments -->
  
  <!-- Flat Dataset args -->
  <arg name="bag_name" default="/home/lukas/Documents/Datasets/flat_dataset/run1.bag"/> 
  <arg name="bag_play_rate" default="1"/>  <!-- Realtime factor -->
  <arg name="use_sim_time" default="true"/>
  <arg name="use_detectron" default="false"/>


  <!-- Run -->
  
  <!-- Play data -->
  <include file="$(find panoptic_mapping_utils)/launch/play_flat_dataset.launch" pass_all_args="true"/>

  <!-- RVIZ Visualization -->
  <node type="rviz" name="rviz_voxblox" pkg="rviz" args="-d $(find panoptic_mapping_ros)/config/rviz/voxblox.rviz"/>
  
  <!-- Voxblox -->
  <node name="voxblox_node" pkg="voxblox_ros" type="tsdf_server" output="screen" args="-alsologtostderr">
    <remap from="pointcloud" to="/data/pointcloud"/>
    <param name="use_tf_transforms" value="true"/>
    <param name="tsdf_voxel_size" value="0.05" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="color_mode" value="color" />
    <param name="update_mesh_every_n_sec" value="3.0" />
    <param name="method" value="fast" />
    <param name="max_ray_length_m" value="5" />
    <param name="allow_clear" value="true" />
    <param name="verbose" value="true" />
    <param name="mesh_filename" value="/home/lukas/Documents/PanopticMapping/Data/voxblox/5cm_vxblx.ply" />
  </node>
  
</launch>
